sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -691,7 +692,8 @@ private[sql] class SQLConf extends Serializable with CatalystConf with Logging {
   def variableSubstituteDepth: Int = getConf(VARIABLE_SUBSTITUTE_DEPTH)
 
   def warehousePath: String = {
-    getConf(WAREHOUSE_PATH).replace("${system:user.dir}", System.getProperty("user.dir"))
+    new File(getConf(WAREHOUSE_PATH).replace("${system:user.dir}",
+      System.getProperty("user.dir"))).toURI.toString
```

Comment:
I am not sure it is the right fix. What will happen if a user's default fs is hdfs or s3 and he/she set `spark.sql.warehouse.dir` in the conf?